lib/ansible/galaxy/api.py

```diff
@@ -46,6 +47,28 @@ def wrapped(*args, **kwargs):
     return wrapped
 
 
+def rate_limit_backoff(function):
+    def run_function(*args, **kwargs):
+        # TODO: support user configuration
+        retries = 10
+        delay = 2
+        max_delay = 60  # Bikeshed
+
+        for retry in range(0, retries):
+            delay = random.randint(0, min(max_delay, delay * 2 ** retry))
+
+            try:
+                return function(*args, **kwargs)
+            except GalaxyError as e:
+                # Ideally, this would handle rate limit errors and not unknown errors.
+                # TODO: Check if v3 also uses 520 error code for rate limit errors, or if that's a v2 quirk.
+                if e.http_code not in (520, 429):
```

Comment:
Today I've also been experiencing sporadic 502 errors.

Reply:
I'd argue that these should be configurable too.

Also, it'd be great to move this to a module-global constant like:
```python
from ansible.module_utils.six.moves import http_client


RETRIABLE_HTTP_CODES = {
    http_client.TOO_MANY_REQUESTS,
    520,  # internal rate limit code?
}
```

It's best to put magic numbers into constants so that when used, they'd add more meaning to the code.